In [1]:
# LOAD ANY REQUIRED LIBRARIES

import pandas as pd
import numpy as np
from datetime import datetime

In [2]:
# SETUP A PATH TO THE DATA

data_path = "./data/drifting_longlines.csv"

In [3]:
# LOAD THE DATA

df_data = pd.read_csv(data_path)

In [7]:
# HAVE A LOOK AT THE DATA STRUCTURE

df_data.head()

vessel_num   timestamp  distance_from_shore  distance_from_port  speed  \
0  1.260000e+13  1327136504          232994.2813         311748.6563    8.2   
1  1.260000e+13  1327136605          233994.2656         312410.3438    7.3   
2  1.260000e+13  1327136734          233994.2656         312410.3438    6.8   
3  1.260000e+13  1327143281          233994.2656         315417.3750    6.9   
4  1.260000e+13  1327143341          233996.3906         316172.5625    6.1   

       course        lat        lon  is_fishing               source  
0  230.500000  14.865583 -26.853662          -1  dalhousie_longliner  
1  238.399994  14.863870 -26.856800          -1  dalhousie_longliner  
2  238.899994  14.861551 -26.860649          -1  dalhousie_longliner  
3  251.800003  14.822686 -26.865898          -1  dalhousie_longliner  
4  231.100006  14.821825 -26.867579          -1  dalhousie_longliner

# Vessels
Let's look at the number of distinct vessels and then the time span covered by the most common vessel.

In [5]:
df_data['vessel_num'].value_counts()

2.380000e+13    236842
1.180000e+14    225289
5.140000e+13    146687
8.130000e+13    100672
2.150000e+14     92842
7.720000e+13     84873
1.040000e+14     74585
2.570000e+14     43554
2.810000e+14     18096
1.680000e+14     13289
1.260000e+13     11846
Name: vessel_num, dtype: int64

We only have 11 distinct vessels in this data (unless the identifiers get recycled)


The most common Vessel (2.380000e+13) in the data appears 236,842 times


It seems like this multiple trips. Lets look at the timestamps to validate this.

In [6]:

df_temp = df_data[df_data['vessel_num']==2.380000e+13]
latest = datetime.fromtimestamp(max(df_temp['timestamp']))
oldest = datetime.fromtimestamp(min(df_temp['timestamp']))
print("Vessel data between ", oldest, " and ", latest )

Vessel data between  2012-02-27 03:23:14  and  2016-11-25 10:59:22


Ok, so this has to be multiple trips, we need to work out how to break these into individual voyages?

We may need to parse the data per vessel and look for long gaps between timestamps in order to generate a voyage identifier for each distinct trip made by that vessel.